Lib/test/test_xml_etree.py

```diff
@@ -2693,6 +2694,41 @@ def __eq__(self, o):
         e.extend([ET.Element('bar')])
         self.assertRaises(ValueError, e.remove, X('baz'))
 
+    def test_remove_with_clear_children(self):
+        # See: https://github.com/python/cpython/issues/126033
+
+        class X(ET.Element):
+            def __eq__(self, o):
+                root.clear()
```

Comment:
What is the difference between `del root[:]` and `root.clear()`?

Reply:
If I replace `del root[:]` by `root.clear()` in the current test `test_remove_with_mutating`, the following crashes (but I forgot to investigate the distinction):

```py
        class X(ET.Element):
            def __eq__(self, o):
                e.clear()
                return False
        e = ET.Element('foo')
        e.extend([X('bar')])
        self.assertRaises(ValueError, e.remove, ET.Element('baz'))

        e = ET.Element('foo')
        e.extend([ET.Element('bar')])
        self.assertRaises(ValueError, e.remove, X('baz'))
```

Reply:
Should `del e[:]` also set `extra` to NULL? (and yes e.clear() sets extra to NULL)